index.js

```diff
@@ -110,66 +78,94 @@ function fastifyCors (fastify, opts, next) {
         )
       }
 
-      next()
+      return next()
     })
   }
 
-  function configureOrigin (req, reply, callback) {
-    const reqOrigin = req.headers.origin
-    if (isOriginFunction) {
-      const result = origin.call(fastify, reqOrigin, _onOrigin)
-      if (result && typeof result.then === 'function') {
-        result.then(res => _onOrigin(null, res), callback)
-      }
-    } else {
-      _configureOrigin(origin)
+  function preflightHandler (req, reply) {
+    // Do not handle preflight requests if the origin was not allowed
+    if (!req.corsOriginAllowed) {
```

Comment:
This is a new behaviour. Also technically the `404` code is not fitting the spec as the ressource is here but the CORS is not allowed. I'd rather use `400`. Returning a `404` would be handled in a custom function.

Reply:
Glad to see an observant eye. This was changed in my last PR #96 . 

This is due to the fact that I moved the preflight logic into the `OPTIONS *` route handler. There was some certain motivation for that (discussed in #96 ) but the general points were to:
1. allow overriding of `OPTIONS` routes (seems to be the motivation behind #88 )
1. make fastify-cors fit better into the fastify plugin paradigm. The previous implementation seemed to be a fairly strict port of express-cors, where middleware can either handle a request or pass it long to the next middleware in the chain. That model doesn't fit with plugins, hooks, and a "static" routing table that fasitfy uses.

Back to the question. This line is to allow for the case when the `origin` option is `false` or is a function that returns `false`. According to the docs, this should disable CORS (and the preflight handler). With middleware, this can just be checked when the middleware executes. With a static route, the route will always exist. 404 is to "simulate" the route not being there.

From a client perspective, if CORS is disabled, shouldn't the `OPTIONS` routes added by fastify-cors be "not found"? 

Let me know if I didn't understand your point. Maybe the name of the field is misleading... perhaps it should be `corsEnabled` similar???

Reply:
I renamed this to `corsPreflightEnabled` which better describes the flag.